README.md

```diff
@@ -112,4 +112,83 @@ argocd login --insecure $ARGOCD_SERVER --username $ARGOCD_USER --password $ARGOC
 
 argocd app delete $app-bootstrap --grpc-web -y
 argocd app list --grpc-web
-```
\ No newline at end of file
+```
+
+
+## Tools
+
+### `app-config.yaml` template
+
+The `template_app_config_local.sh` is a tool to help in the generation of 
+ the `app-config.yaml` file. It contains several resources that can be either 
+ enabled or not depending on the provided information.
+
+The goal of the bash script is to transform the provided parameters into 
+ `jinja2` variables that will be replaced inside the script.
+
+The script prompts variables to the user upon execution, 
+ **unless they are defined using the inputs of the script**.
+
+**NOTE**: The script is executed in _strict_ mode so all the 
+ variables that are defined in the template and don't have a _default_ 
+ value must be provided.
+
+| Variable | Meaning |
+| --- | --- |
+| `--batch` | Execute in batch mode. Skips the parameter prompt. |
+| `--help` | Show the script help message. |
+| `--template_file` | Custom location for the template file. |
+| `--verbosity` | Verbosity level for the script output. |
+
+When called without the batch mode the input parameters will be requested to the user, if not already defined.
+
+
+```bash
+./template_app_config_local.sh
+```
+
+```
+Template file location (default: ./manifest/app-config.local.yaml.j2): 
+Backstage APP Base URL (default: localhost:3000): 
+Backstage Backend Base Host (default: localhost): 
+Backstage Backend Base Port (default: 7007): 
```

Comment:
IT is not needed to ask 3 times the same question ;-) Either the user will say `localhost` OR will provide an URL `https://idp-backstage.apps.qshift.snowdrop.dev`

Reply:
The value of the listen port is required to be added to the yaml template. I think we should keep requesting it separately.

```yaml
backend:
  baseUrl: {{ BACKSTAGE_BACKEND_BASE_HOST | default('localhost') }}:{{ BACKSTAGE_BACKEND_BASE_PORT | default('7007') }}
  listen:
    port: {{ BACKSTAGE_BACKEND_BASE_PORT | default('7007') }}
```

Reply:
I dont think that most of the users will change the default ports (3003 = front & 7007 = backend) when using backstage locally. 

If you want to nevertheless handle the ports, then we could do this:

- Backstage url (default: localhost):
- Backstage frontend application port (default: 3003):
- Backstage backend application port (default: 7007):

Reply:
IIUC, the backstage base URL will always be the same for the frontend and the backend. I'll make this change.